### PR TITLE
Added c Sudoku solver

### LINKED ISSUE #1099: Add C sudoku solver
I saw [this](https://github.com/OpenGenus/cosmos/blob/master/code/backtracking/sudoku_solve/SudokuSolve.cpp).

I'd like to add my C version of that problem.

### PR DESCRIPTION
**Pull request by ReimuA**
Resolved #1099.

### CODE REVIEW

File: code/backtracking/sudoku_solve/SudokuSolve.c [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#include <stdlib.h>
+#include <stdbool.h>
+
+static const int map_size = 9;
+
+/*
+** Check a number's existance on a line.
+*/
+
+int		exist_on_line(char map[map_size][map_size + 2], const int pos_y, const char c)
```
Comment (hwchiu) [inline code/backtracking/sudoku_solve/SudokuSolve.c]: Could you check the coding style for all functions? such as indent?
Thanks

File: code/backtracking/sudoku_solve/SudokuSolve.c [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+			return (1);
+	}
+	return (0);
+}
+
+/*
+** Check a number's existance on a column.
+*/
+
+int		exist_on_column(char map[map_size][map_size + 2], const int pos_x, const char c)
```
Comment (hwchiu) [inline code/backtracking/sudoku_solve/SudokuSolve.c]: Indent here

File: code/backtracking/sudoku_solve/SudokuSolve.c [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+{
+	int	i = 0;
```
Comment (hwchiu) [inline code/backtracking/sudoku_solve/SudokuSolve.c]: Indent here

File: code/backtracking/sudoku_solve/SudokuSolve.c [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+			return (1);
+	}
+	return (0);
+}
+
+/*
+** Check a number's existance on a block.
+*/
+
+int		exist_on_block(char map[map_size][map_size + 2], const int pos_x, const int pos_y, const char c)
```
Comment (hwchiu) [inline code/backtracking/sudoku_solve/SudokuSolve.c]: Indent here

File: code/backtracking/sudoku_solve/SudokuSolve.c [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+/*
+** Solve the sudoku by using backtracking.
+** 
+** The following could be improved by changing
+** the order of completion instead of going
+** from 0 to map_size².
+*/
+
+int 	resolve_sudoku(char map[map_size][map_size + 2], const int pos)
```
Comment (hwchiu) [inline code/backtracking/sudoku_solve/SudokuSolve.c]: Indent here

File: code/backtracking/sudoku_solve/SudokuSolve.c [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+	map[pos_y][pos_x] = '0';
+	return (0);
+}
+
+/*
+** Read from stdin.
+*/
+
+int 	read_map(char map[map_size][map_size + 2])
```
Comment (hwchiu) [inline code/backtracking/sudoku_solve/SudokuSolve.c]: Indent here

File: code/backtracking/sudoku_solve/SudokuSolve.c [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+ 	{
+ 		if (!fgets(map[i], map_size + 2, stdin) ||
+ 			(int)strlen(map[i]) < map_size)
+ 			return (1);
+ 		map[i++][map_size] = '\0';
+ 	}
+ 	return (0);
+}
+
+int		main(void)
```
Comment (hwchiu) [inline code/backtracking/sudoku_solve/SudokuSolve.c]: Indent here

File: code/backtracking/sudoku_solve/SudokuSolve.c [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+{
+	int		i = 0;
```
Comment (hwchiu) [inline code/backtracking/sudoku_solve/SudokuSolve.c]: Indent here

**Comment by ReimuA** [comment]
My bad, fixed weird indent.

**Comment by AdiChat** [comment]
Looks good 👍 

Hope you are enjoying your journey with **OpenGenus**

**Pull request merged**